File: backend/internal/models/reward.go
package models

import "time"

type Reward struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	PointsCost  int64     `json:"points_cost"`
	Inventory   int       `json:"inventory"`
}

type Redemption struct {
        ID        uint      `gorm:"primaryKey" json:"id"`
        CreatedAt time.Time `json:"created_at"`
        UserID    uint      `json:"user_id"`
        RewardID  uint      `json:"reward_id"`
        Status    string    `json:"status"`
        Reward    Reward    `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;" json:"reward"`
}

File: mobile/lib/models/reward.dart
class Reward {
  final int id;
  final String name;
  final String description;
  final int pointsCost;
  final int inventory;

  Reward({
    required this.id,
    required this.name,
    required this.description,
    required this.pointsCost,
    required this.inventory,
  });

  factory Reward.fromJson(Map<String, dynamic> json) {
    return Reward(
      id: json['id'] as int,
      name: json['name'] as String? ?? '',
      description: json['description'] as String? ?? '',
      pointsCost: (json['points_cost'] as num?)?.toInt() ?? 0,
      inventory: json['inventory'] as int? ?? 0,
    );
  }
}
